refactor use of container struct from daemon

MHBauer opened the pull request:
- do existence check instead of get container
- new connect method on daemon.

like #16966

@calavera @duglin @icecrime 

The existence check on the outside could be considered redundant.

duglin commented:
I'd prefer to remove the daemon.Exists() check since the function you're calling will do the exact same check (and return the exact same error) immediately anyway. I think it only makes sense to do this check in the API layer in cases where don't want to enter the daemon/impl layer for some reason (like there might be unwanted side-effects).

MHBauer commented:
Updates based on comments, and a basic integration test.

duglin commented:
LGTM if janky ever does its stuff :-)

duglin commented:
ping @jfrazelle @vdemeester @runcom

MHBauer committed:
refactor use of container struct from daemon

 - do existence check instead of get container
 - new connect method on daemon.
 - cli network disconnect integration test

Signed-off-by: Morgan Bauer <mbauer@us.ibm.com>

MHBauer commented:
rebased. builds were all working before the rebase.

MHBauer commented:
of course. I try to add a test, and the test doesn't compile.

duglin commented:
No good deed...  :-)

duglin commented:
LGTM

ping @vdemeester @jfrazelle

vdemeester commented on integration-cli/docker_cli_network_unix_test.go:
```diff
@@ -135,6 +136,11 @@ func (s *DockerNetworkSuite) TestDockerNetworkCreateDelete(c *check.C) {
 	assertNwNotAvailable(c, "test")
 }
 
+func (s *DockerSuite) TestDockerNetworkDeleteNotExists(c *check.C) {
```
I'm wondering why this test is needed ? Probably because to validate the refactor still works, but this test should have been add when pushing the `docker network` subcommand, right ? (I ask the question to avoid, or not, conflicts :stuck_out_tongue: )

duglin replied:
I assumed he noticed this test was missing while touching/testing the networking API changes. I did a quick check and didn't notice this being tested.

MorganBauer replied:
Yea, thought it would be nice to have, and was an easy gap to cover.

vdemeester replied:
Fair engough :wink:

vdemeester commented:
Small question, other than that, looks good :wink:

vdemeester commented:
LGTM 🐰
